Source/WebCore/platform/mediastream/ios/MediaCaptureStatusBarManager.h

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (C) 2022 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#if ENABLE(MEDIA_STREAM) && PLATFORM(IOS_FAMILY)
+
+#include <wtf/CompletionHandler.h>
+#include <wtf/FastMalloc.h>
+#include <wtf/RetainPtr.h>
+#include <wtf/WeakPtr.h>
+
+OBJC_CLASS WebCoreMediaCaptureStatusBarHandler;
+
+namespace WebCore {
+
+class MediaCaptureStatusBarManager : public CanMakeWeakPtr<MediaCaptureStatusBarManager> {
+    WTF_MAKE_FAST_ALLOCATED;
+public:
+    using TapCallback = Function<void(CompletionHandler<void()>&&)>;
+    using ErrorCallback = Function<void()>;
+    MediaCaptureStatusBarManager(TapCallback&& callback, ErrorCallback&& errorCallback)
+        : m_tapCallback(WTFMove(callback))
+        , m_errorCallback(WTFMove(errorCallback))
```

Comment:
`WebCoreMediaCaptureStatusBarHandler` is a heavyweight class we definitely don't want to leak, so it might be worth adding a destructor that asserts `m_handler` is null to catch future errors in debug builds, and calls `stop()` just in case.

Reply:
I simplified a bit the class and added some additional ASSERTs